presentation/src/main/java/com/bunbeauty/presentation/view_model/menu/edit_menu_product/EditMenuProductViewModel.kt

```diff
@@ -0,0 +1,208 @@
+    private val updateMenuProductUseCase: UpdateMenuProductUseCase
+) : BaseViewModel() {
+
+    private val mutableState = MutableStateFlow(EditMenuProductDataState())
+    val editMenuProductUiState = mutableState.mapToStateFlow(viewModelScope) { state ->
+        mapState(state)
+    }
+
+    fun loadData(productUuid: String) {
+        viewModelScope.launchSafe(
+            block = {
+                val menuProduct = getMenuProductUseCase(productUuid)
+                mutableState.update { oldState ->
+                    oldState.copy(
+                        menuProduct = menuProduct,
+                        name = menuProduct.name,
+                        constName = menuProduct.name,
+                        description = menuProduct.description,
+                        newPrice = menuProduct.newPrice.toString(),
+                        oldPrice = menuProduct.oldPrice?.toString() ?: "",
+                        state = EditMenuProductDataState.State.SUCCESS,
+                        nutrition = menuProduct.nutrition?.toString() ?: "",
+                        utils = menuProduct.utils ?: "",
+                        isVisible = menuProduct.isVisible,
+                    )
+                }
+            },
+            onError = {
+                mutableState.update { oldState ->
+                    oldState.copy(
+                        state = EditMenuProductDataState.State.ERROR
+                    )
+                }
+            }
+        )
+    }
+
+    fun updateMenuProduct() {
+        viewModelScope.launchSafe(
+            block = {
+                mutableState.update { oldState ->
+                    oldState.copy(
+                        isLoadingButton = true,
+                        hasNameError = false,
+                        hasDescriptionError = false,
+                        hasNewPriceError = false,
+                        hasOldPriceError = false,
+                        hasNutritionError = false,
+                    )
+                }
+
+                val name = mutableState.value.name.trim()
+                mutableState.value.menuProduct?.let { menuProduct ->
+                    updateMenuProductUseCase(
+                        menuProduct = menuProduct.copy(
+                            name = name,
+                            description = mutableState.value.description.trim(),
+                            newPrice = mutableState.value.newPrice.toIntOrNull() ?: 0,
+                            oldPrice = mutableState.value.oldPrice
+                                .ifEmpty { null }
+                                ?.toInt() ?: 0,
+                            utils = if (mutableState.value.nutrition.isEmpty()) {
+                                mutableState.value.utils
+                            } else {
+                                ""
+                            },
+                            nutrition = mutableState.value.nutrition.toIntOrNull(),
```

Comment:
Null nutrition will not update value, lets also add `?: 0` and make a rework on backend